sdk/resources/mgmt/src/main/java/com/azure/management/RestClientBuilder.java

```diff
@@ -117,8 +117,9 @@ public RestClientBuilder withBaseUrl(AzureEnvironment environment, AzureEnvironm
      * @param credential the credential.
      * @return builder
      */
-    public RestClientBuilder withCredential(TokenCredential credential) {
+    public RestClientBuilder withCredential(AzureTokenCredential credential) {
```

Comment:
Does this means all other credential e.g. (`ManagedIdentityCredential`) need to be wrapped in `AzureTokenCredential`?

What will be the `domain` or `clientId` if this is the case (another credential wrapped)?

Not sure about it. Why do we had to use a subclass instead of a widely used base class `TokenCredential`?